### PR TITLE
Define InitWrapper message

### PR DESCRIPTION
**Pull request by tiziano88**
# Checklist

- [ ] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by
        [Cloudbuild](/cloudbuild.yaml)
  - [ ] I have updated [documentation](/docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: oak_io/src/lib.rs [hunk 1]
```diff
@@ -83,14 +83,105 @@ impl<L: Decodable, R: Decodable> Decodable for Either<L, R> {
...
+}
+
+/// Implementation of [`Encodable`] for [`InitWrapper`] that encodes the handle of the command
+/// receiver channel in the first handle of the resulting [`Message`].
+impl<Init: Encodable, Command: Encodable + Decodable> Encodable for InitWrapper<Init, Command> {
+    fn encode(&self) -> Result<Message, OakError> {
+        let mut init_message = self.init.encode()?;
+        init_message
+            .handles
+            .insert(0, self.command_receiver.handle.handle);
```
Comment (ipetr0v) [inline oak_io/src/lib.rs:102]: Is `0` added here because we encode an `Either`?
Please add a comment

File: oak_io/src/lib.rs [hunk 2]
```diff
@@ -83,14 +83,105 @@ impl<L: Decodable, R: Decodable> Decodable for Either<L, R> {
     }
 }
 
+/// A wrapper struct that holds an init message, plus the handle of a channel from which to read
+/// command messages. This is useful for patterns in which an Oak node needs some data at
+/// initialization time, but then handles commands of a different type in an event-loop pattern.
```
Comment (ipetr0v) [inline oak_io/src/lib.rs:88]: Does it mean that `Init` and `Commands` are now sent in separate channels?

**Review by ipetr0v — APPROVED** [review]

File: oak_io/src/lib.rs (hunk 1 above)
Comment (tiziano88) [reply oak_io/src/lib.rs:105]: The 0 is just the position in which to insert (i.e. at the beginning). Added comment to clarify.

File: oak_io/src/lib.rs (hunk 2 above)
Comment (tiziano88) [reply oak_io/src/lib.rs]: Yes, clarified. The point is that you can have a router node initialize the worker node with some information, and then send it a channel (coming from somewhere else) from which to read commands (e.g. gRPC invocations). I think this should help with the aggregator and trusted database pattern, let's try and use it and see if it makes things better.

**Review by tiziano88 — COMMENTED** [review]
Thanks for the quick review.

**Pull request merged**